### DISCUSSION TITLE
[FEATURE] Request for Documentation on Building Release Version without LeakCanary

### DISCUSSION THREAD

**Discussion by ghost** [zeapo/Android-Password-Store#2414] [Q&A]
Dear project maintainer,

Thank you for your reply regarding the inclusion of LeakCanary in the snapshot builds of the Android Password Store. As you mentioned, it is intentional for testing purposes, but I would like to request documentation on building a release version without LeakCanary.

I understand that the default release configuration does not include LeakCanary, but it would be helpful to have clear instructions on how to exclude it from the build process.

Would it be possible for you to provide documentation or guidance on how to build a release version without LeakCanary? Any help or information you can provide would be greatly appreciated.

Thank you for your time and attention to this matter!

**Comment by msfjarvis** [comment]
I'm not going to give you from scratch instructions, you'll need to figure basic setup yourself if you're determined to build the app. The APS specific instructions you'll need are for selecting the build flavor, which is described [here](https://docs.passwordstore.app/docs/users/build-types); and for signing release builds, which I'll explain now.

APS reads the credentials for your keystore from a `keystore.properties` file located at the project root. The following 4 keys are expected in that file, along with their values which are omitted here for obvious reasons. The `storeFile` parameter is a relative path to the keystore file from the project root, everything else is self-explanatory.

```properties
# keystore.properties
keyAlias=
keyPassword=
storeFile=
storePassword=
```

I'll reiterate my first point to make it clearer: I will not hand hold you through basic Android Studio setup or debug your build errors if you make a mistake. Every bit of information that is specific to APS is included in this comment, and you should not expect any further assistance past that.

**Reply by msfjarvis** [reply]
I haven't received any meaningful contributions to the app in nearly two years and I have made my peace with that continuing to be the case till I give up on the project. If your first interaction with me starts with stating blatant falsehoods about the dependency stack I use then you can't fault me for my skepticism.

**Comment by msfjarvis** [comment]
Also to be clear, "This would be beneficial for users who prefer minimalistic software and for those who have privacy and security concerns with the inclusion of a library like LeakCanary." is provably false and misinformed. LeakCanary is a thoroughly vetted and widely trusted open source library, which does not create any privacy or security issues by virtue of its use.

**Comment by ghost** [comment]
I've tried following the steps you suggested, but LeakCanary still exists in the release APK build.

**Reply by msfjarvis** [reply]
LeakCanary is not active in the default configuration. It is included because it is allowed to be enabled at runtime through the debug logging switch and with the build-time configuration that enables debugging by default, which snapshots use.

**Reply by ghost** [reply]
`./gradlew assembleFreeRelease` **still** produces an APK build with LeakCanary enabled.

**Reply by msfjarvis** [reply]
I don't see anything on my end that suggests LeakCanary is enabled without the debug option turned on but I've made the lifecycle handling more explicit regardless https://msfjarvis.dev/aps/ddd958a8d604. If you still find LeakCanary to be dumping the heap in your build, I'll take a PR.

**Reply by ghost** [reply]
I always build the Android Password Store with its latest source code, but the APK build **still** contains the LeakCanary utility. Particularly, the launcher icon of Leaks is showed after I've installed the build, and the Password Store is **getting crashed** now.

**Reply by msfjarvis** [reply]
The icon won't go away because as I already said, LeakCanary is still being bundled just without heap dumping enabled. The app doesn't crash for me, and without any logs I can't really do anything anyway so you'll have to either attach a log of the crash here or raise a PR with a fix.

**Reply by ghost** [reply]
How can I get a build without the LeakCanary library and the "Leaks" launcher icon?

**Reply by msfjarvis** [reply]
Revert ddd958a8d6042a2db61caceedd2eebea928c2b90, 9156a81ad772a70198737e84ea9f7a06870b43dc, and 8e3b4422a933f87c1a36a6efd583ebf0803891a1.

**Reply by msfjarvis** [reply]
I've spent entirely too much time on your meaningless crusade against LeakCanary so I'm going to go ahead and close this discussion.
